src/ai/backend/manager/repositories/model_serving/repository.py

```diff
@@ -964,3 +972,39 @@ async def _do_mutate() -> MutationResult:
             raise
         except Exception:
             raise
+
+    @model_serving_repository_resilience.apply()
+    async def search_auto_scaling_rules_validated(
+        self,
+        querier: BatchQuerier,
+        user_id: uuid.UUID,
+        user_role: UserRole,
+        domain_name: str,
+    ) -> EndpointAutoScalingRuleListResult:
```

Comment:
I'm not sure if the user id, role, and domain information should be included.
I was hoping for this information to be passed as conditions. (I'm thinking of keeping the implementation of the repository layer clean and just passing different values from above. (The deeper you go into the internal layers, the fewer if statements or match cases there should be.))